### ISSUE TITLE
Skip To Content button always visible

### ISSUE THREAD

**Issue by jongund** [w3c/apg-redesign#7]
I would like to have the "Skip To Content" link always visible, for example centered at the top of the page.

This would make the typically "hidden" accessibility information of headings and landmarks visible to all users.

The benefits of this approach are:
* It increases the awareness to everyone the existence of headers and landmarks on a web page and they can be used for page navigation by all users.
* Authors will have an incentive to be more careful in defining landmarks and headings if they know that everyone will be able to see and use them.

**Comment by a11ydoer** [comment]
I agree with you @jongund 
This change will also meet WCAG 2.2 visible control 3.2.7.

**Comment by shawna-slh** [comment]
The main WAI website has "Skip to Content" link always visible in a strip across the top of pages. Visually it is the first thing in a right-aligned list of links, so sometimes it is near the middle of the page, or sometimes further left or right.
![image](https://user-images.githubusercontent.com/6991814/160019857-4ce93d65-cf64-4b36-ab3c-ecc29102cd4f.png)

APG redesign is using WAI's “minimal header design” template that has a minimal header without that strip. The template has “Skip to Content” in a large yellow button that appears on focus:
![image](https://user-images.githubusercontent.com/6991814/160019903-88d8b85d-8fb2-4680-98e9-b6a58da9fc36.png)
This is acceptable under [SC 3.2.7](https://w3c.github.io/wcag/understanding/visible-controls.html): “… features such as skip links… are not intended to be covered by this criterion.”

It seems that it would be good communicate the importance of being careful in defining landmarks and headings where they are addressed. I’m missing how having an always visible "Skip To Content" would help that? And, I actually worry it might “backfire” with authors who feel that it always visible is inelegant and cluttering?

**Comment by jongund** [comment]
@shawna-slh @mcking65 
It looks like the APG redesign team has missed the mark on integrating the "Skip To Content" link into the APG template.  The button should not stick out like a sore thumb and cover other content on the page.  The button should be integrated into the overall look and feel of the website and be the first focusable element on the page.   It should be easy to identify and be available to anyone to use to navigate the structure of the page, including people with disabilities.  This type of implementation is exactly the opposite of how the "Skip To Content" link should be implemented if you want other websites to consider using this type of feature to make their websites more accessible.

**Comment by richnoah** [comment]
@jongund thank you for your insight.  But to be clear the APG Redesign is utilizing a WAI Template.  We are not integrating anything into the template.  There are current discussions on how the template might utilize some of the designs proposed.  

@shawna-slh we are noticing that the ARIA Practices headline text is now present.  Is there any active template work underway?

**Comment by shawna-slh** [comment]
First, let me clarify that the minimal header design screenshot above has been refined separately from the APG redesign team (so don't blame them for that :-). We are now working on merging our designs (and things may be a bit wonky during that time).

Having said that, it looks like they also used a Skip to Contents that is only visible on focus:
![image](https://user-images.githubusercontent.com/6991814/160169423-cc4ab0a7-2987-402e-bcbe-f59b8a5792e2.png)

@jongund This is not "set in stone". We can change it if we decide to.

My understanding is that a visible-only-on-focus Skip to Content is OK accessibility and **generally the recommended approach in most cases**. It seems this minimal header design is one of those case. I welcome additional information.

**Comment by jongund** [comment]
@shawna-slh @richnoah 
The importance of having "skip to content" being visible is a filtered set of landmarks and headings is available to everyone for navigation and also allows people to get a high level view the structure of the page without having to scroll through it.  

A visible "Skip to Content" button has many benefits, including:

1. **Educational:** People who probably never would think about landmarks and headings will now have an opportunity to learn about them when the activate the "Skip to Content" button.  It helps raise awareness of basic accessibility features of a website, by making a filtered set of landmarks and headings  available to everyone, not just users of screen readers and people who primarily use the keyboard.  For accessibility to advance more people need to learn about landmark and heading concepts that are typically hidden from most users.  Unfortunately browsers do not provide any functionality for users to benefit from proper use of landmarks and headings, so it is up to authors to help educate the general public of these basic concepts and I hope the WAI website redesign sees the value of this important opportunity.
2.  **Authoring:** Authors and content management systems can easily test changes in content to make sure the filtered set of landmarks and headings make sense and will learn to pay more attention landmarks and headings if they know anyone might be using them, not just people using screen readers and keyboard mostly users. 
3. **Users:** The "Skip To Content" script provides a new and modern way for all users to get an overview of a web page and navigate to content on the page.    Screen reader users also get a filtered set of landmarks and headings, rather than the entire lists provided by screen readers.  Keyboard mostly users get more choices than just the main content. 

While making "Skip To Main" link a "popup" is the most popular technique and generally accepted practice for accessibility, I don't consider the technique a best practice, especially in light of newer technologies like "Skip To Content", which when used properly can help ALL users understand and navigate to content on a web page, but again only if they are given an affordance to know that it is an option (e.g. a visible "Skip To Content" button).
